adapter/internal/discovery/xds/server.go

```diff
@@ -466,6 +520,109 @@ func UpdateAPI(vHost string, apiProject mgw.ProjectAPI, deployedEnvironments []*
 		orgIDOpenAPIRoutesMap[organizationID] = routesMap
 	}
 
+	apiName := mgwSwagger.GetTitle()
+	apiVersion := mgwSwagger.GetVersion()
+	apiRangeIdentifier := GenerateIdentifierForAPIWithoutVersion(vHost, apiName)
+
+	_, _, _, err = ValidateAndGetVersionComponents(apiVersion)
```

Comment:
Shall we do this inside the if block `if isIntelligentRoutingEnabled` and reuse the returned values instead of calling it multiple times?

Reply:
Maybe we can return a struct pointer from this function. For example. And we can reuse this value in other functions.

```go
type struct SemVersion {
	Major int
	Minor int
	Patch int
}

func ValidateAndGetVersionComponents(version string) (*SemVersion, error) {
}
```

Reply:
> Shall we do this inside the if block `if isIntelligentRoutingEnabled` and reuse the returned values instead of calling it multiple times?

+1 to move this inside `if isIntelligentRoutingEnabled` block and have another if block to check whether the version is compatible with semantic versioning.
Note that the `ValidateAndGetVersionComponents` function is used on different versions such as current api version, current latest major range version and current latest minor range version. Therefore, we can't reuse the returned values.

Reply:
> Maybe we can return a struct pointer from this function. For example. And we can reuse this value in other functions.
> 
> ```go
> type struct SemVersion {
> 	Major int
> 	Minor int
> 	Patch int
> }
> 
> func ValidateAndGetVersionComponents(version string) (*SemVersion, error) {
> }
> ```

+1, with Patch is being an optional field

Reply:
Got your point. What if we change the type of the internal memory map to the following?
```go
orgIDLatestAPIVersionMap map[string]map[string]map[string]SemVersion
```
So we do not need to recompute this value again.

Reply:
This will introduce another overhead of concatenating and building the version from the version components again.

We may be able to overcome these computing overheads by updating the `SemVersion` to hold the exact version and the version components as well.

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/3376/commits/bc8484ac9a3f5629075cd867af5e0659ecf3b80f, https://github.com/wso2/product-microgateway/pull/3376/commits/f1b404e5a34f6b004ac91f88dd4af63ff7bf9644 and https://github.com/wso2/product-microgateway/pull/3376/commits/10575c0498ec6037c245cc1473891adaad8ee27a